src/status_im/transport/message/v1/protocol.cljs

```diff
@@ -0,0 +1,105 @@
+(ns status-im.transport.message.v1.protocol
+  (:require [status-im.utils.config :as config]
+            [status-im.constants :as constants]
+            [status-im.chat.core :as chat]
+            [status-im.transport.message-cache :as message-cache]
+            [status-im.transport.db :as transport.db]
+            [status-im.transport.core :as transport]
+            [status-im.transport.message.core :as message]
+            [status-im.transport.utils :as transport.utils]))
+
+(def ^:private whisper-opts
+  {:ttl       10 ;; ttl of 10 sec
+   :powTarget config/pow-target
+   :powTime   config/pow-time})
+
+(defn init-chat
+  ([chat-id cofx]
+   (init-chat chat-id (transport.utils/get-topic chat-id) cofx))
+  ([chat-id topic {:keys [db] :as cofx}]
+   {:db (assoc-in db [:transport/chats chat-id] (transport.db/create-chat topic))}))
+
+(defn is-new? [message-id]
+  (when-not (message-cache/exists? message-id)
+    (message-cache/add! message-id)))
+
+#_(defn requires-ack [message-id chat-id {:keys [db] :as cofx}]
```

Comment:
Not clear why these are commented out, either let them stay (fine if no one is calling them) or remove